frontend/src/components/common/formQuestions/QuestionsAccordionItem.tsx

```diff
@@ -9,52 +9,91 @@ import {
   Text,
 } from "@chakra-ui/react";
 
-import { FormQuestion } from "../../../types/CampsTypes";
+import { CreateFormQuestion, FormQuestion } from "../../../types/CampsTypes";
 import QuestionCard from "./QuestionCard";
+import RegistrationQuestionCard from "../../pages/CampCreation/RegistrationForm/RegistrationQuestionCard";
+import NoRegistrationQuestionsCard from "../../pages/CampCreation/RegistrationForm/NoRegistrationQuestionsCard";
 
 type QuestionsAccordionItemProps = {
   fixedQuestions: Array<FormQuestion>; // A question for presentational purposes only, Admin can not edit or delete this
   dynamicQuestions: Array<FormQuestion>; // Questions added by the admin during the respective workflow which can be editted + deleted
   accordionTitle: string;
+  onDeleteCustomQuestion?: (questionToBeDeleted: CreateFormQuestion) => void;
+  onEditCustomQuestion?: (
+    oldQuestion: CreateFormQuestion,
+    newQuestion: CreateFormQuestion,
+  ) => void;
+  isTemplatePage?: boolean;
 };
 
 const QuestionsAccordionItem = ({
   fixedQuestions,
   dynamicQuestions,
   accordionTitle,
+  onDeleteCustomQuestion,
+  onEditCustomQuestion,
+  isTemplatePage = true,
 }: QuestionsAccordionItemProps): React.ReactElement => {
+  const fixedQuestionsCount = fixedQuestions.length;
+  const dynamicQuestionsCount = dynamicQuestions.length;
+
   return (
     <AccordionItem border="none" marginBottom="40px">
       <AccordionButton
-        minH="50px"
-        backgroundColor="primary.green.600"
         borderRadius="20px"
+        bg="primary.green.600"
         _hover={{ bg: "primary.green.600" }}
       >
         <Box flex="1" textAlign="left">
-          <Text color="text.white.100" fontWeight="bold" size="25px">
+          <Text color="text.white.100" textStyle="displayLarge" p={2}>
             {accordionTitle}
           </Text>
         </Box>
         <AccordionIcon color="white" />
       </AccordionButton>
 
-      <AccordionPanel>
-        {fixedQuestions.map((question, index) => (
-          <QuestionCard
-            question={question}
-            viewOnly
-            key={`${question.question}_${index + 1}`}
-          />
-        ))}
-        {dynamicQuestions.map((question, index) => (
-          <QuestionCard
-            question={question}
-            viewOnly={false}
-            key={`${question.question}_${index + fixedQuestions.length + 1}`}
-          />
-        ))}
-      </AccordionPanel>
+      {isTemplatePage ? (
+        <AccordionPanel>
+          {fixedQuestions.map((question, index) => (
+            <QuestionCard
```

Comment:
Why not just conditionally render the `QuestionCard` or `RegistrationQuestionCard` here? 
so everything would remain the same but you would have:
```
{fixedQuestions.map((question, index) => (
          { isTemplatePage ? ( 
    <QuestionCard
            question={question}
            viewOnly
            key={`${question.question}_${index + 1}`}
          />) : (<RegistrationQuestionCard ... > )
))}
```

Reply:
This way, we would need two ternary operators instead of one- the first ternary for fixed questions and the second ternary for dynamic questions. It's also easier to account for the fact that the Registration Form has a `<NoRegistrationQuestionsCard>` case.